### PR TITLE
Search if address bar input is not a hostname

### PR DESCRIPTION
**Pull request by karmanyaahm**
Implemented issue #3 
While I would've preferred not to add another dependency, the string_validator library had the right options and seemed to have a lot of logic in it. I think the basic inputs work, but could you test more possible edge cases for the address bar input?

I also moved some things around in the settings page to make the code more organized with multiple settings.

### CODE REVIEW

File: lib/main.dart [hunk 1]
```diff
@@ -181,7 +181,9 @@ class AppState extends State<App> with AutomaticKeepAliveClientMixin {
     feeds = (prefs.getStringList('feeds') ?? []);
     settings = {
       "homepage":
-          (prefs.getString("homepage") ?? "gemini://gemini.circumlunar.space/")
+          (prefs.getString("homepage") ?? "gemini://gemini.circumlunar.space/"),
+      "search":
+          (prefs.getString("homepage") ?? "gemini://gus.guru/search")
```
Comment (snoe) [inline lib/main.dart:186]: I think this is a mistake?

File: lib/address_bar.dart [hunk 2]
```diff
@@ -38,11 +38,32 @@ class AddressBar extends StatelessWidget {
...
+                              })) {
+                            newURL = Uri.tryParse(value);
+                            if (!newURL.hasScheme) {
+                              newURL = Uri.parse("gemini://" + value);
+                            }
+                          } else {
+                            String searchEngine =
+                                (await SharedPreferences.getInstance())
+                                    .getString("search");
+                            newURL = Uri.parse(searchEngine);
```
Comment (snoe) [inline lib/address_bar.dart:63]: When I first ran and typed in a search this threw a null pointer exception, going into settings and pressing enter on the search field fixed it.

```
E/flutter (22250): [ERROR:flutter/lib/ui/ui_dart_state.cc(177)] Unhandled Exception: NoSuchMethodError: The getter 'length' was called on null.
E/flutter (22250): Receiver: null
E/flutter (22250): Tried calling: length
E/flutter (22250): #0      Object.noSuchMethod (dart:core-patch/object_patch.dart:51:5)
E/flutter (22250): #1      Uri.parse (dart:core/uri.dart:792:17)
E/flutter (22250): #2      AddressBar.build.<anonymous closure>
package:deedum/address_bar.dart:63
E/flutter (22250): <asynchronous suspension>
E/flutter (22250): #3      AddressBar.build.<anonymous closure> (package:deedum/address_bar.dart)
package:deedum/address_bar.dart:1
E/flutter (22250): #4      EditableTextState._finalizeEditing
package:flutter/…/widgets/editable_text.dart:1819
E/flutter (22250): #5      EditableTextState.performAction
package:flutter/…/widgets/editable_text.dart:1688
E/flutter (22250): #6      TextInput._handleTextInputInvocation
package:flutter/…/services/text_input.dart:1168
E/flutter (22250): #7      MethodChannel._handleAsMethodCall
package:flutter/…/services/platform_channel.dart:430
E/flutter (22250): #8      MethodChannel.setMethodCallHandler.<anonymous closure>
package:flutter/…/services/platform_channel.dart:383
E/flutter (22250): #9      _DefaultBinaryMessenger.handlePlatformMessage
package:flutter/…/services/binding.dart:283
E/flutter (22250): #10     _invoke3.<anonymous closure> (dart:ui/hooks.dart:280:15)
```

**Comment by snoe** [comment]
@karmanyaahm I'm not sure if you saw the feed reader yet, but `about:feeds` should go to the user's feed gmi feed. On this branch it goes to the search provider.

**Comment by snoe** [comment]
I fixed the issues I noted, thanks for this!

**Comment by karmanyaahm** [comment]
Thanks for fixing it.
